src/partition/repart.c

```diff
@@ -4149,6 +4220,39 @@ static int run(int argc, char *argv[]) {
         if (r < 0)
                 return r;
 
+        if (arg_image) {
+                assert(!arg_root);
+
+                /* Strictly open this in read-only fashion: we shall modify the partition table, not the file systems */
```

Comment:
"Stricly read-only" and "modify the partition table" seem somewhat contradictory ;)